### PR TITLE
Add onlyOwner and gameReady modifier into Hangman

### PR DESCRIPTION
**Pull request by ejwessel**
## Description 
- Anybody can read game state
- only the owner can play
- only when the game is ready can it be played

## Changes
- Added tests

### CODE REVIEW

**Comment by ejwessel** [comment]
@brianwchou  I still need to update the factory unit tests

File: test/Hangman.test.js [hunk 1]
```diff
@@ -23,235 +19,284 @@ contract('Hangman', async (accounts) => {
...
+              await truffleAssert.passes(
+                  hangmanContract.makeWordGuess(web3.fromAscii("hello")),
+                  "Transaction failed");
+
+              await truffleAssert.reverts(
+                  hangmanContract.makeWordGuess(web3.fromAscii("testing")),
+                  "solution is found");
+          });
+      });
+      });
```
Comment (brianwchou) [inline test/Hangman.test.js]: indent here is off

File: contracts/Hangman.sol [hunk 2]
```diff
@@ -15,14 +15,20 @@ contract Hangman is Ownable {
...
+    function setSolution(bytes _solution, uint _maxAllowedMisses) onlyOwner external {
+        require(solution.length == 0, "solution already set");
         solution = _solution;
         maxAllowedMisses = _maxAllowedMisses;
-        currentMisses = 0;
+    }
+
+    modifier gameReady() {
+        require(solution.length > 0, "solution not set");
+        _;
```
Comment (brianwchou) [inline contracts/Hangman.sol]: what is this line for?

Comment (ejwessel) [reply contracts/Hangman.sol]: the line is what allows it to be a modifier.

**Comment by ejwessel** [comment]
update url usage and the constructor parameter

**Pull request merged**